From 1b33eb3f589cc110522b6a0da7e9c4eeb40cbdd8 Mon Sep 17 00:00:00 2001
From: Kyle Steere <kyle.steere@chainguard.dev>
Date: Mon, 13 Jan 2025 07:47:28 -0600
Subject: [PATCH] fix: requests patch

fixes GHSA-pr98-23f8-jwxv and GHSA-6v67-2wr5-gvf4

Signed-off-by: Kyle Steere <kyle.steere@chainguard.dev>

fixed GHSA-4g8c-wm8x-jfhw

Signed-off-by: Kyle Steere <kyle.steere@chainguard.dev>

fix GHSA-4g8c-wm8x-jfhw, GHSA-pq2g-wx69-c263

Signed-off-by: Kyle Steere <kyle.steere@chainguard.dev>
---
 build.gradle      | 11 +++++++++--
 gradle.properties |  7 ++++++-
 2 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/build.gradle b/build.gradle
index d9d4fa7a..f8107bb8 100644
--- a/build.gradle
+++ b/build.gradle
@@ -49,6 +49,11 @@ configurations.all {
         force("com.fasterxml.jackson.core:jackson-databind:" + jacksonVersion)
         force("com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:" + jacksonVersion)
         force("com.fasterxml.jackson.module:jackson-module-scala_" + kafkaScalaVersion + ":" + jacksonVersion)
+        force("io.vertx:vertx-core:" + vertxVersion)
+        force("org.apache.commons:commons-compress:" + commonsCompressVersion)
+        force("io.netty:netty-handler:" + nettyVersion)
+        force("org.eclipse.jetty:jetty-http:" + jettyHttpVersion)
+        force("net.minidev:json-smart:2.5.2")
     }
 }

@@ -93,7 +98,7 @@ dependencies {
     implementation("io.micronaut:micronaut-http-server-netty")
     implementation("io.micronaut:micronaut-jackson-databind")
     implementation("io.swagger.core.v3:swagger-annotations")
-    runtimeOnly("ch.qos.logback:logback-classic")
+    runtimeOnly("ch.qos.logback:logback-classic:" + logbackVersion)
     runtimeOnly("org.yaml:snakeyaml")
     implementation("io.micronaut.security:micronaut-security-annotations")
     implementation("io.micronaut.security:micronaut-security-jwt")
@@ -136,7 +141,7 @@ dependencies {
     implementation 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310'

     // avro
-    implementation 'org.apache.avro:avro:1.11.3'
+    implementation 'org.apache.avro:avro:1.12.0'

     // jackson-module-scala
     implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-scala_2.13', version: '2.17.1'
@@ -173,6 +178,8 @@ dependencies {
     implementation group: 'io.projectreactor', name: 'reactor-core', version: '3.6.7'

     implementation 'io.jsonwebtoken:jjwt-impl:0.12.6'
+
+    implementation 'io.netty:netty-common:4.1.118.Final'
 }

 /**********************************************************************************************************************\
diff --git a/gradle.properties b/gradle.properties
index ae531b3b..7a70b71d 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -3,4 +3,9 @@ confluentVersion=7.4.4
 kafkaVersion=3.6.2
 kafkaScalaVersion=2.13
 lombokVersion=1.18.32
-jacksonVersion=2.17.1
\ No newline at end of file
+jacksonVersion=2.17.1
+logbackVersion=1.5.16
+commonsCompressVersion=1.26.0
+vertxVersion=4.4.8
+nettyVersion=4.1.118.Final
+jettyHttpVersion=12.0.12
--
2.43.0
